src/CMakeLists.txt

```diff
@@ -113,43 +188,45 @@ if(CCTAG_WITH_CUDA)
 
   if(CCTAG_HAVE_SHFL_DOWN_SYNC)
     target_compile_definitions(CCTag PRIVATE "-DCCTAG_HAVE_SHFL_DOWN_SYNC")
-  endif(CCTAG_HAVE_SHFL_DOWN_SYNC)
+  endif()
 
   if(CCTAG_NO_THRUST_COPY_IF)
     target_compile_definitions(CCTag PRIVATE "-DCCTAG_NO_THRUST_COPY_IF")
   endif()
 
+
   # This is nececessary for the CCTagConfig.cmake to correctly export the
   # includes, always because we used CUDA_ADD_LIBRARY
   set_target_properties(CCTag PROPERTIES INTERFACE_INCLUDE_DIRECTORIES
-       "$<INSTALL_INTERFACE:$<INSTALL_PREFIX>/include>;${CUDA_INCLUDE_DIRS};${Boost_INCLUDE_DIRS};${Eigen_INCLUDE_DIR};${OpenCV_INCLUDE_DIRS}")
-  get_target_property(testprop CCTag INTERFACE_INCLUDE_DIRECTORIES )
-  message(STATUS "testprop: ${testprop}")
+          "$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}>;$<BUILD_INTERFACE:${generated_dir}>;$<INSTALL_INTERFACE:$<INSTALL_PREFIX>/include>;${CUDA_INCLUDE_DIRS};${Boost_INCLUDE_DIRS};${OpenCV_INCLUDE_DIRS}")
+  # Just for testing
```

Comment:
remove trailing comments

Reply:
I left those lines commented because they are useful to debug, `cuda_add_library` does not support well all the PRIVATE / PUBLIC dependencies and those lines are useful to see if everything is ok

Reply:
ok